application/migrations/005_create_logging_table.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+defined('BASEPATH') OR exit('No direct script access allowed');
+
+class Migration_create_logging_table extends CI_Migration
+{
+
+    public function up()
+    {
+        // Create new table for each new day
+        $this->db->query('
+        CREATE TABLE dashboard.ci_logs (
```

Comment:
:+1: 

Is going to cause problems since the table _does_ exist for any existing apps (staging and production)? Unless, you're configuring it to use a new table name, is that why it's `dashboard.ci_logs`? I don't see a change to CILogger that would tell it to use this new table, so it's unclear to me. 

Should the `dashboard.` prefix come from a variable? (I feel like I've seen a table_prefix setting somewhere...

Having you considered using an `IF NOT EXIST` kind of thing instead?

Reply:
The table prefix came from watching what it actually created just-in-time in the code that was removed.

Doh, I absolutely should have remembered that this table will exist in existing environments! Great catch. I will add an `IF NOT EXIST`.

Reply:
re: the prefix... could this be the database name? I feel like the table name should just be `ci_logs`. The database name is probably different between local development and production.

Reply:
You're right, it's the database name. [You can see `CILogger` using it in the `INSERT` here.](https://github.com/GSA/project-open-data-dashboard/blob/7f63fa2c89766deb04ec8038a65fd08f2b5c640e/application/libraries/CILogger.php#L99-L100) I didn't catch that `CILogger` was using that; I thought that it was some implicit schema-creation thing in MySQL. How silly... I'll remove it, and just have the migration create `ci_logs`. I'll see what I can do to make the migration of data into that new table from the existing ones painless on us.

Reply:
👍 don't spend time on migrating the data. That table is using the [`ARCHIVE` engine](https://dev.mysql.com/doc/refman/5.7/en/archive-storage-engine.html), which is unindexed. Would be a pain to have to query.

In the past, this table has grown so big it just brings the database to a [halt](https://github.com/GSA/datagov-deploy/issues/535). And then we just drop the table (and let it re-create itself). I'm not sure why the table even really exists, maybe for auditing, but it was designed in a way that it would be impossible to actually query it for anything.

Arghh, but now it's part of the migrations and won't automatically re-create itself. Maybe we need more thought here, I'm not sure if we should remove this code, or do something else with it.

Reply:
Oh, but also in terms of migrating data, there's no action to take. If the table exists, the data is exactly where it should be :)

Reply:
I learned a few things in the course of making these last few commits.

First, the default schema name for a MySQL database is the same as the database name. So when you create a table named `<dbname>.ci_logs`, the resulting table will be called `ci_logs` and queryable as both `<dbname>.ci_logs` and `ci_logs`. This means all the existing tables auto-created in the other environments should already be called just `ci_logs`. (So, another reason there's no need to migrate data.)

Second, I think the `CILogger.php` class was explicitly including the schema in the table name on its `INSERT`s to be extra careful to make sure it's logging in the default schema. This defends against causing errors in the remote case that the app it's being used in has set another schema in the course of doing its own thing.

Finally, I noticed there was a comment I'd copied over above the just-in-time table creation code in `CILogger.php` that said:
> // Create new table **for each new day**.

I think there was likely a cronjob that renamed the DB table each day, and nuked old tables after a certain number of days to avoid filling up the database. That's actually a little clever and makes me feel like moving the table creation to a migration may have been a mistake, but on the other hand there's no sign of any code that actually does the rotation so we should expect to see behavior that looks exactly like: 

> this table has grown so big it just brings the database to a halt. And then we just drop the table (and let it re-create itself). 

So... 
1. I can leave it as-is, and feel good that code isn't creating tables inline.
2. I could put it back the way it was, creating the table just-in-time, accepting the bad smell.
3. I could both put it back **and** try to recreate the missing cronjob to rename the table daily, which would address https://github.com/GSA/datagov-deploy/issues/535 .
4. I could just remove the CILogger code as you suggested.

Right now I'm inclined to leave it as-is since figuring out the final dispensation of query-logging code is out of scope for what the PR is trying to do and I want to move on with changes that are going to give us confidence/reduce friction.

Reply:
(Will wait to resolve this part of your review once you have a chance to agree/disagree.)

Reply:
(Aaron [concurred below](https://github.com/GSA/project-open-data-dashboard/pull/218#discussion_r356740996).)